target/linux/generic/hack-5.15/290-net-dsa-mv88e6xxx-allow-built-in-when-PTP-disabled.patch

```diff
@@ -0,0 +1,44 @@
+From 04cdbf10182b4cb9726c67ad7ae8d05d0d6bed95 Mon Sep 17 00:00:00 2001
+From: "Johnny S. Lee" <foss@jsl.io>
+Date: Thu, 15 Dec 2022 17:49:04 +0800
+Subject: [PATCH] net: dsa: mv88e6xxx: allow built-in when PTP disabled
+
+Objects of PTP hardware timestamping are not linked when the support is
+disabled, therefore MV88E6xxx should not be prevented to be built-in
+unless PTP support is enabled.
+
+Fixes: e5f31552674e ("ethernet: fix PTP_1588_CLOCK dependencies")
+Signed-off-by: Johnny S. Lee <foss@jsl.io>
+---
+ drivers/net/dsa/mv88e6xxx/Kconfig | 13 +++++++++++--
+ 1 file changed, 11 insertions(+), 2 deletions(-)
+
+--- a/drivers/net/dsa/mv88e6xxx/Kconfig
++++ b/drivers/net/dsa/mv88e6xxx/Kconfig
+@@ -2,7 +2,7 @@
+ config NET_DSA_MV88E6XXX
+ 	tristate "Marvell 88E6xxx Ethernet switch fabric support"
+ 	depends on NET_DSA
+-	depends on PTP_1588_CLOCK_OPTIONAL
++	depends on NET_DSA_MV88E6XXX_PTP_OPTIONAL
```

Comment:
If there already is a separate config option for PTP support in mv88e6xxx, shouldn't be the following enough:

```suggestion
-	depends on PTP_1588_CLOCK_OPTIONAL
+	depends on PTP_1588_CLOCK if NET_DSA_MV88E6XXX_PTP
```

assuming the PTP parts are properly guarded in the driver, and don't use code from `PTP_1588_CLOCK` anyway.

Then `NET_DSA_MV88E6XXX` wouldn't depend on `PTP_1588_CLOCK` regardless if it's m or y, unless you explicitly enable `NET_DSA_MV88E6XXX_PTP`.

And you wouldn't need `NET_DSA_MV88E6XXX_PTP_OPTIONAL` at all (or any of the other changes).

Reply:
I experienced the hell of "Kconfig recursive dependency" a bit before the initial patch came to shape, and I can tell you `depends on <expr> if <expr>` is not a valid statement in Kconfig language.